students/brandon_nguyen/session07/html_render.py

```diff
@@ -23,7 +23,7 @@ def append(self, new_content):
 
     def render(self, out_file, cur_ind=""):
         # loop through the list of contents + recursive render(...)
-        # TODO how to be pure if ELEMENT has abstract_tag instead.
+        # TODO how to be pure if ELEMENT has abstract_tag instead?
```

Comment:
I missed this comment earlier, but I'm not sure what it means to be "pure"...? What is the `abstract_tag` for, in the first place?